orderer/common/multichannel/follower_resources.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright IBM Corp. 2017 All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package multichannel
+
+import (
+	"github.com/hyperledger/fabric-protos-go/common"
+	"github.com/hyperledger/fabric/bccsp"
+	"github.com/hyperledger/fabric/common/channelconfig"
+	"github.com/hyperledger/fabric/common/ledger/blockledger"
+	"github.com/hyperledger/fabric/common/policies"
+	"github.com/hyperledger/fabric/internal/pkg/identity"
+	"github.com/hyperledger/fabric/orderer/consensus"
+	"github.com/hyperledger/fabric/protoutil"
+	"github.com/pkg/errors"
+)
+
+// FollowerResources encapsulates all the resources the follower needs to do its job:
```

Comment:
I realize this is an iterative change, so I understand where this is coming from.  But let's not copy the monolithic interface style from the consenter package, or the embedding of the ledger resources to implement things.

We can have a plain struct with fields each satisfying an interface (or, better yet, as a literal if we can get away with it), that way we can more sanely compose things and it's clearer where our implementations are coming from.

Reply:
Yes, I agree completely. The "support" pattern is a bad one and I am eager to get rid of it as soon as possible (at least for the follower at this point in time) . I will prepare the ground in this PR and complete that in the following PRs.

Reply:
I plan to do this in the next task: https://jira.hyperledger.org/browse/FAB-18070

Reply:
Great, thanks.